common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -0,0 +1,814 @@
+            List<FormattedCharSequence> lines = font.split(note.right(), 200);
+            int offsetY = -(font.lineHeight * lines.size()) / 2;
+            for (FormattedCharSequence line : lines) {
+                int offsetX = -font.width(line) / 2;
+                font.drawInBatch(line, offsetX, offsetY, color, false, pose, source, false, 0x80000000, 0xf000f0);
+                offsetY += font.lineHeight + 2;
+            }
+            poseStack.popPose();
+        }
+    }
+
+    private static void renderChests(
+            PoseStack poseStack,
+            LootrunInstance lootrun,
+            int color,
+            MultiBufferSource.BufferSource source,
+            long chunkLong) {
+        VertexConsumer consumer = source.getBuffer(RenderType.lines());
+        Set<BlockPos> chests = lootrun.chests().get(chunkLong);
+
+        float red = ((float) FastColor.ARGB32.red(color)) / 255;
+        float green = ((float) FastColor.ARGB32.green(color)) / 255;
+        float blue = ((float) FastColor.ARGB32.blue(color)) / 255;
+
+        for (BlockPos chest : chests) {
+            LevelRenderer.renderLineBox(poseStack, consumer, new AABB(chest), red, green, blue, 1f);
+        }
+
+        source.endBatch();
+    }
+
+    private static void renderPoints(
+            PoseStack poseStack,
+            MultiBufferSource.BufferSource source,
+            Long2ObjectMap<List<List<ColoredPoint>>> points,
+            long chunkLong) {
+        List<List<ColoredPoint>> locations = points.get(chunkLong);
+
+        Level level = McUtils.mc().level;
+        if (level == null) return;
+
+        for (List<ColoredPoint> locationsInRoute : locations) {
+            VertexConsumer consumer = source.getBuffer(LootrunRenderType.LOOTRUN_LINE);
+            Matrix4f lastMatrix = poseStack.last().pose();
+            boolean sourceBatchEnded = false;
+
+            List<ColoredPoint> toRender = new ArrayList<>();
+
+            boolean disablePoint = false;
+            BlockPos lastBlockPos = null;
+
+            for (ColoredPoint point : locationsInRoute) {
+                boolean pauseDraw = false;
+                BlockPos blockPos = new BlockPos(point.vec3());
+
+                if (blockPos.equals(lastBlockPos)) { // Do not recalculate block validness
+                    if (disablePoint) {
+                        pauseDraw = true;
+                    } else if (!toRender.isEmpty()) {
+                        toRender.add(point);
+                    }
+                } else {
+                    Iterable<BlockPos> blocks = getBlocksForPoint(point);
+
+                    boolean barrierInArea = false;
+                    boolean validBlock = false;
+
+                    for (BlockPos blockInArea : blocks) {
+                        BlockState blockStateInArea = level.getBlockState(blockInArea);
+                        if (blockStateInArea.is(Blocks.BARRIER)) {
+                            barrierInArea = true;
```

Comment:
`barrierInArea` check is a bit iffy, given that there exists barriers in use in SE